sdk/planetarycomputer/azure-planetarycomputer/samples/planetary_computer_01_ingestion_management.py

```diff
@@ -0,0 +1,416 @@
+        client.ingestion.delete_source(id=source.id)
+        logging.info(f"Deleted existing source: {source.id}")
+
+    # Create connection info with managed identity
+    connection_info = ManagedIdentityConnection(container_uri=container_uri, object_id=managed_identity_object_id)
+
+    # Create ingestion source with unique ID
+    source_id = str(uuid.uuid4())
+    ingestion_source = ManagedIdentityIngestionSource(id=source_id, connection_info=connection_info)
+    created_source = client.ingestion.create_source(body=ingestion_source)
+    logging.info(f"Created managed identity ingestion source: {created_source.id}")
+
+    # List managed identities
+    logging.info("Listing available managed identities:")
+    managed_identities = list(client.ingestion.list_managed_identities())
+    for identity in managed_identities:
+        logging.info(f"  - Object ID: {identity.object_id}")
+        logging.info(f"    Resource ID: {identity.resource_id}")
+
+    return source_id
+
+
+def create_or_replace_source(
+    client: PlanetaryComputerProClient, sas_container_uri: str, sas_token: str, source_id: str
+):
+    """Demonstrate create_or_replace_source idempotent operation.
+
+    This assumes the source already exists (created by create_sas_token_ingestion_source).
+    It demonstrates that create_or_replace_source can be called multiple times with the same source_id
+    to update/replace the source (in this case, updating the SAS token).
+    """
+    # Validate required parameters
+    if not sas_container_uri:
+        raise ValueError(
+            "AZURE_INGESTION_SAS_CONTAINER_URI environment variable must be set for create_or_replace_source"
+        )
+    if not sas_token:
+        raise ValueError("AZURE_INGESTION_SAS_TOKEN environment variable must be set for create_or_replace_source")
+
+    # Create connection info with SAS token
+    connection_info = SharedAccessSignatureTokenConnection(
+        container_uri=sas_container_uri, shared_access_signature_token=sas_token
+    )
+
+    # Create ingestion source
+    ingestion_source = SharedAccessSignatureTokenIngestionSource(id=source_id, connection_info=connection_info)
+
+    # First call - replaces the existing source with original token
+    logging.info(f"First call to create_or_replace_source with existing source ID: {source_id}")
+    first_result = client.ingestion.replace_source(id=source_id, body=ingestion_source)
+    logging.info(f"First call result: {first_result.id}")
+    # Second call - replaces again with modified token (demonstrates update capability)
+    # Generate a valid SAS token format with required fields: permission, start, and expiration
+    from datetime import datetime, timedelta
+
+    start_time = datetime.utcnow().strftime("%Y-%m-%dT%H:%M:%SZ")
+    expiry_time = (datetime.utcnow() + timedelta(days=7)).strftime("%Y-%m-%dT%H:%M:%SZ")
+    updated_token = f"sp=rl&st={start_time}&se={expiry_time}&sv=2023-01-03&sr=c&sig=UpdatedRandomSignature123456"
```

Comment:
If this is code that customers will actually need to write - we should look into whether we could provide some kind of helper function - formatting datetimes is kinda painful.